### PR TITLE
Switch to hardhat-deploy & Update frontend contract-interactions

### LINKED ISSUE #37: Use hardhat-deploy for contract deployments
We currently use a self-made script for contract deployment and saving of deployed addresses to files for the frontend. We now found the feature rich hardhat plugin https://github.com/wighawag/hardhat-deploy which does this, including etherscan contract validation. So we should switch to using it.

### PR DESCRIPTION
**Pull request by wottpal**
Resolves #37

* Switched to `hardhat-deploy`
  * See https://github.com/yieldgate/yieldgate/issues/37#issuecomment-1199875269 for detailed change description
* chore: Renamed package `hardhat` → `contracts`
* chore: Updated environment variables in `frontend` & `contracts` (esp. RPC URLs)
* contracts: Added Goerli deployment
* frontend: Moved typechain-types back to `contracts` package and import them from there
* frontend: Updated contract interactions in frontend to comply with interface changes from #45
  * Fetch pool address from factory first and then perform stake, unstake, claim actions there 
  * Added "Deploy Pool" action if no pool was found

### CODE REVIEW

**Review by sebastianst — CHANGES_REQUESTED** [review]
Looks great! Only some minor improvement proposals.

File: packages/contracts/deploy/001_deploy_yieldgate.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    throw new Error(`No deployer address defined for network ${hre.network.name}`)
+  }
+  console.log(`Deploying as ${deployer}…`)
+
+  const aave = AaveAddresses[hre.network.name]
+  if (!aave) {
+    throw new Error(`No Aave addresses for network ${hre.network.name}`)
+  }
+
+  await deploy('YieldGate', {
```
[the remark this answers is not in the document]
Comment (wottpal) [reply packages/contracts/deploy/001_deploy_yieldgate.ts:24]: Typescript doesn't allow me to do so. The `DeployFunction` has the following return type: `Promise<void | boolean>`. Just FYI by returning nothing in an async function right now it's implicitly `Promise<void>` at the moment.

[I found this example](https://github.com/wighawag/template-ethereum-contracts/blob/96e7e12528dfeefd167cdf78fd0700286b068ccd/deploy/002_deploy_greetings_registry.ts#L19) out there, but I'm not sure what they are achieving by this (but also didn't spend too much time studying the docs).

[the remark this answers is not in the document]
Comment (sebastianst) [reply packages/contracts/deploy/001_deploy_yieldgate.ts:24]: ok thanks, was just curious.

File: packages/contracts/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+node_modules
+.env
+cache
+artifacts
+deployments/localhost
+coverage
+coverage.json
```
[the remark this answers is not in the document]
Comment (wottpal) [reply packages/contracts/.gitignore]: See https://github.com/yieldgate/yieldgate/pull/49/commits/01e90d50c51e05a1c464525a1311eb5077727a45 where I'm improving all the ignore files you are mentioning. But please, let us only leave the `.env` in the hardhat `.gitignore`. Long story, but I already lost some money by leaking a private key... 🙈

If you are d'accord please also resolve the other comments.

[the remark this answers is not in the document]
Comment (sebastianst) [reply packages/contracts/.gitignore]: haha sure thing 💸

File: packages/frontend/src/components/creator/CreatorCardActions.tsx [hunk 3]
```diff
@@ -49,35 +53,96 @@ export const CreatorCardActions: FC<CreatorCardActionsProps> = ({
...
+      setPoolAddress(undefined)
+      return
+    }
+    const factoryContract = new ethers.Contract(
+      contracts.YieldGate.address,
+      contracts.YieldGate.abi,
+      provider
+    ) as YieldGate
+    const poolAddress = await factoryContract.beneficiaryPools(creator.address)
+    if (!poolAddress || BigNumber.from(poolAddress).eq(0)) {
```
[the remark this answers is not in the document]
Comment (wottpal) [reply packages/frontend/src/components/creator/CreatorCardActions.tsx]: Awesome, thank you! Actually, I even googled for something like this during implementation but couldn't find something. Updated in https://github.com/yieldgate/yieldgate/pull/49/commits/31dc08c64b13434447915d69911482512f5271cd

File: packages/frontend/src/components/creator/CreatorCardActions.tsx [hunk 4]
```diff
@@ -158,27 +228,33 @@ export const CreatorCardActions: FC<CreatorCardActionsProps> = ({
...
-    )?.[0] as ClaimedEvent
-    const claimedAmount = ethers.utils.formatUnits(claimedEvent?.args?.amount, 'finney')
-    console.log({ claimedEvent, claimedAmount })
+    let claimedAmount
+    try {
+      const poolContract = new ethers.Contract(
+        poolAddress,
+        BeneficiaryPoolAbi.abi,
+        signer
+      ) as BeneficiaryPool
```
[the remark this answers is not in the document]
Comment (wottpal) [reply packages/frontend/src/components/creator/CreatorCardActions.tsx]: Thanks for bringing this up. I updated the syntax everywhere in https://github.com/yieldgate/yieldgate/pull/49/commits/9fdd4281286e03c189fc4ba4a3570055a813af95

**Review by sebastianst — APPROVED** [review]

**Pull request merged**